mesonbuild/modules/i18n.py

```diff
@@ -14,47 +14,243 @@
-        '--keyword=_',
-        '--keyword=N_',
-        '--keyword=C_:1c,2',
-        '--keyword=NC_:1c,2',
-        '--keyword=g_dcgettext:2',
-        '--keyword=g_dngettext:2,3',
-        '--keyword=g_dpgettext2:2c,3',
-
-        '--flag=N_:1:pass-c-format',
-        '--flag=C_:2:pass-c-format',
-        '--flag=NC_:2:pass-c-format',
-        '--flag=g_dngettext:2:pass-c-format',
-        '--flag=g_strdup_printf:1:c-format',
-        '--flag=g_string_printf:2:c-format',
-        '--flag=g_string_append_printf:2:c-format',
-        '--flag=g_error_new:3:c-format',
-        '--flag=g_set_error:4:c-format',
-    ]
+        'keywords' : [
+            '_',
+            'N_',
+            'C_:1c,2',
+            'NC_:1c,2',
+            'g_dcgettext:2',
+            'g_dngettext:2,3',
+            'g_dpgettext2:2c,3',
+        ],
+
+        'flags': [
+            'N_:1:pass-c-format',
+            'C_:2:pass-c-format',
+            'NC_:2:pass-c-format',
+            'g_dngettext:2:pass-c-format',
+            'g_strdup_printf:1:c-format',
+            'g_string_printf:2:c-format',
+            'g_string_append_printf:2:c-format',
+            'g_error_new:3:c-format',
+            'g_set_error:4:c-format',
+        ],
+    },
 }
 
+class I18nPotGeneratorHolder(InterpreterObject):
+
+    def __init__(self, generator):
+        super().__init__()
+        self.held_object = generator
+
+        self.methods.update({'process': self.process_method,
+                             'add_files': self.add_files_method,
+                             'merge_files': self.merge_files_method,
+                             'add_merge_files': self.add_merge_files_method,
+                             })
+
+    def add_files_method(self, args, kwargs):
+        self.held_object.add_files(args, kwargs)
+
+    def merge_files_method(self, args, kwargs):
+        return self.held_object.merge_files(args, kwargs)
+
+    def add_merge_files_method(self, args, kwargs):
+        if len(args) != 1:
+            raise coredata.MesonException('add_merge_files requires at least one positional argument (file).')
+        if 'extra_args' in kwargs:
+            raise coredata.MesonException('add_merge_files does not accept extra_args.')
+        self.held_object.add_files(args, kwargs.copy())
+        return self.held_object.merge_files(args, kwargs.copy())
+
+    def process_method(self, args, kwargs):
+        targets = self.held_object.process(args, kwargs)
+        self.held_object.interpreter.process_new_values(targets)
+
+
+class I18nPotGenerator:
+
+    @staticmethod
+    def _get_keywords(kwargs, preset):
+        keywords = mesonlib.stringlistify(kwargs.pop('keywords', []))
+        for k in preset['keywords']:
+            if k not in keywords:
+                keywords.append(k)
+        return keywords
+
+    def _get_preset(self, kwargs):
+        preset = kwargs.pop('preset', self.preset)
+        if not preset in PRESET_ARGS:
```

Comment:
__[flake8]__
* [E713] test for membership should be 'not in'

<!-- Comment by [SideCI](https://sideci.com) -->